### PR TITLE
DALA-1863 P2P framework backend

### PR DESCRIPTION
**Pull request by MarcTM01**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one E2E Test exists testing the new feature
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] The new version is deployed with fake data to the dev server "dev1" using this branch
  - [x] Run with setting `Use real data` turned off and `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server
  - [x] All implemented Social Logins have been tested manually in the UI
- [x] The new version is deployed with real data to a dev server using this branch
  - [x] Run with both settings `Reset non-user related Docker Volumes & Re-populate` and `Use real data` turned on
  - [x] It's verified that the CD run is green
- [x] If any work on the UI is to be merged, those changes were also documented in the Figma

### CODE REVIEW

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/P2p.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.junit.jupiter.api.Test
+
+class P2p {
+    private val apiAccessor = ApiAccessor()
+
+    private val listOfOneP2pDataSet = apiAccessor.testDataProviderForP2pData.getTData(1)
+    private val listOfOneCompanyInformation = apiAccessor.testDataProviderForP2pData
+        .getCompanyInformationWithoutIdentifiers(1)
+
+    private fun getP2pDatasetWithSortedSectors(dataset: PathwaysToParisData): PathwaysToParisData {
```
Comment (SiegfriedSobkowiak) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/P2p.kt:15]: Wir brauchen das sortiert damit der Vergleich später funktuniert, richtig?

File: dataland-frontend/tests/e2e/fixtures/p2p/P2pDataFixtures.ts [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+/**
+ * Generates a random P2P dataset
+ * @param undefinedProbability the ratio of fields to be undefined (number between 0 and 1)
+ * @returns a random P2P dataset
+ */
+export function generateP2pData(undefinedProbability = 0.5): PathwaysToParisData {
+  const sectors = faker.helpers.arrayElements(Object.values(P2pSector));
+
+  const p2pFixture: PathwaysToParisData = {
```
Comment (SiegfriedSobkowiak) [inline dataland-frontend/tests/e2e/fixtures/p2p/P2pDataFixtures.ts:293]: Könnte man auch gleich zurückgeben ohne eine neue Variable einzuführen.

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/P2p.kt (hunk 1 above)
Comment (marcelrasbachdfine) [reply dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/P2p.kt:15]: Marc hat es im Vergleich sortiert, damit hat es dann funktioniert, da es keine List und kein enum ist.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/enums/p2p/P2pSector.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * An enum for company sectors in the P2P Framework
+ */
+@Schema(
+    enumAsRef = true,
+)
+enum class P2pSector {
+    Ammonia,
+    Automotive,
+    HVCPlastics,
```
[the remark this answers is not in the document]
Comment (marcelrasbachdfine) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/enums/p2p/P2pSector.kt:14]: meinst du alphabetisch sortieren?

[the remark this answers is not in the document]
Comment (marcelrasbachdfine) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/enums/p2p/P2pSector.kt]: habe es jetzt alphabetisch sortiert

**Pull request merged**